### ISSUE TITLE
OpenID Connect Login error (tested with Rocket.Chat v3.1.1) in 1.16.0-RC.1: The property/field [state] does not exist in the class [class io.fusionauth.app.action.oauth2.TokenAction]

### ISSUE THREAD

**Issue by ceefour** [FusionAuth/fusionauth-issues#595]
### Description 
OpenID Connect for Rocket.Chat throws error, was fine in 1.15.8 but cannot downgrade as well due to database schema changes.

### Steps to reproduce
1. Setup OpenID Connect in Rocket.Chat
2. Login via FusionAuth

### Expected behavior
Logged in as usual.

### Screenshots

FusionAuth logs:

```
May 01, 2020 11:09:27.692 AM ERROR io.fusionauth.app.primeframework.error.ExceptionExceptionHandler - An unhandled exception was thrown
org.primeframework.mvc.parameter.el.MissingPropertyExpressionException: While evaluating the expression [state]. The property/field [state] does not exist in the class [class io.fusionauth.app.action.oauth2.TokenAction]
	at org.primeframework.mvc.parameter.el.MemberAccessor.<init>(MemberAccessor.java:84)
	at org.primeframework.mvc.parameter.el.Expression.next(Expression.java:147)
	at org.primeframework.mvc.parameter.el.Expression.traverseToEndForSet(Expression.java:113)
	at org.primeframework.mvc.parameter.el.DefaultExpressionEvaluator.setValue(DefaultExpressionEvaluator.java:131)
	at org.primeframework.mvc.parameter.DefaultParameterHandler.setValues(DefaultParameterHandler.java:201)
	at org.primeframework.mvc.parameter.DefaultParameterHandler.handle(DefaultParameterHandler.java:119)
	at org.primeframework.mvc.parameter.DefaultParameterWorkflow.perform(DefaultParameterWorkflow.java:54)
	at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
	at org.primeframework.mvc.parameter.DefaultURIParameterWorkflow.perform(DefaultURIParameterWorkflow.java:102)
	at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
	at org.primeframework.mvc.scope.DefaultScopeRetrievalWorkflow.perform(DefaultScopeRetrievalWorkflow.java:58)
	at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
	at org.primeframework.mvc.message.DefaultMessageWorkflow.perform(DefaultMessageWorkflow.java:45)
	at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
	at io.fusionauth.app.primeframework.FrontEndTenantWorkflow.perform(FrontEndTenantWorkflow.java:71)
	at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
	at org.primeframework.mvc.action.DefaultActionMappingWorkflow.perform(DefaultActionMappingWorkflow.java:126)
	at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
	at org.primeframework.mvc.workflow.StaticResourceWorkflow.perform(StaticResourceWorkflow.java:97)
	at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
	at org.primeframework.mvc.parameter.RequestBodyWorkflow.perform(RequestBodyWorkflow.java:89)
	at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
	at org.primeframework.mvc.security.DefaultSavedRequestWorkflow.perform(DefaultSavedRequestWorkflow.java:57)
	at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
	at io.fusionauth.app.primeframework.CORSFilter.handleNonCORS(CORSFilter.java:291)
	at io.fusionauth.app.primeframework.CORSFilter.doFilter(CORSFilter.java:252)
	at io.fusionauth.app.primeframework.CORSRequestWorkflow.perform(CORSRequestWorkflow.java:50)
	at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
	at io.fusionauth.app.primeframework.FusionAuthMVCWorkflow.perform(FusionAuthMVCWorkflow.java:88)
	at org.primeframework.mvc.workflow.DefaultWorkflowChain.continueWorkflow(DefaultWorkflowChain.java:44)
	at org.primeframework.mvc.servlet.FilterWorkflowChain.continueWorkflow(FilterWorkflowChain.java:50)
	at org.primeframework.mvc.servlet.PrimeFilter.doFilter(PrimeFilter.java:82)
	at com.inversoft.maintenance.servlet.MaintenanceModePrimeFilter.doFilter(MaintenanceModePrimeFilter.java:59)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at com.inversoft.servlet.UTF8Filter.doFilter(UTF8Filter.java:27)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:199)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:543)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:615)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:818)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1623)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.base/java.lang.Thread.run(Thread.java:832)
```

Rocket.Chat logs:

```
{"line":"392","file":"oauth_server.js","message":"Error in OAuth Server: Failed to complete OAuth handshake with fusionauth at https://login.lovia.life/oauth2/token. failed [500] {\"generalErrors\":[{\"code\":\"[Exception]\",\"message\":\"FusionAuth encountered an unexpected error. Please contact support for assistance.\"}]}","time":{"$date":1588331367776},"level":"warn"}
Exception while invoking method 'login' Error: Failed to complete OAuth handshake with fusionauth at https://login.lovia.life/oauth2/token. failed [500] {"generalErrors":[{"code":"[Exception]","message":"FusionAuth encountered an unexpected error. Please contact support for assistance."}]}
    at CustomOAuth.getAccessToken (app/custom-oauth/server/custom_oauth_server.js:233:18)
    at Object.handleOauthRequest (app/custom-oauth/server/custom_oauth_server.js:290:26)
    at OAuth._requestHandlers.<computed> (packages/oauth2/oauth2_server.js:10:33)
    at middleware (packages/oauth/oauth_server.js:161:5)
    at /app/bundle/programs/server/npm/node_modules/meteor/promise/node_modules/meteor-promise/fiber_pool.js:43:40
```

### Platform

- FusionAuth 1.16.0-RC.1
- Kubernetes 1.16
- MySQL 8
 - Device: Desktop (but independent)
 - OS: Ubuntu 18.04 (but independent)
 - Browser + version: Chrome 81

### Additional context

Currently using database search, not Elasticsearch. But was using Elasticsearch in 1.15.8 . This is an upgraded instance from 1.15.8 (somewhat unintentionally: https://github.com/FusionAuth/fusionauth-issues/issues/563#issuecomment-621369278).

**Comment by ceefour** [comment]
Since this instance is needed for daily operations, I think I will restore MySQL backup and rollback to 1.15.8.

**Comment by robotdan** [comment]
This will be fixed in the final release of 1.16.0. If you do need to use version `1.16.0-RC.1`, the work around is to modify the `fusionauth.properties` file and change 

```
fusionauth.runtime-mode=development
```

to

```
fusionauth.runtime-mode=production
```

or set environment variable to `FUSIONAUTH_RUNTIME_MODE=production`.
